src/mmrelay/db_utils.py

```
@@ -585,13 +584,20 @@ def get_db_path() -> str:
+    Close the given DatabaseManager if provided.
 
-    Closes the manager when non-None; any exception raised by the manager's close() is ignored.
+    If a manager is supplied, calls its close() method; re-raises KeyboardInterrupt and SystemExit, and logs then suppresses any other exceptions.
     """
     if manager:
-        with contextlib.suppress(Exception):
+        try:
             manager.close()
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)